src/es/plottwistnofansub/src/eu/kanade/tachiyomi/extension/es/plottwistnofansub/PlotTwistNoFansub.kt

```diff
@@ -0,0 +1,210 @@
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.network.interceptor.rateLimitHost
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.FormBody
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.toResponseBody
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import org.jsoup.nodes.Entities
+import org.jsoup.select.Elements
+import uy.kohesive.injekt.injectLazy
+
+class PlotTwistNoFansub : ParsedHttpSource() {
+
+    override val name = "Plot Twist No Fansub"
+
+    override val baseUrl = "https://www.plotfansub.com"
+
+    override val lang = "es"
+
+    override val supportsLatest = false
+
+    private val json: Json by injectLazy()
+
+    override val client: OkHttpClient = network.client.newBuilder()
+        .rateLimitHost(baseUrl.toHttpUrl(), 1)
+        .build()
+
+    override fun headersBuilder(): Headers.Builder = Headers.Builder()
+        .add("Referer", "$baseUrl/")
+
+    override fun popularMangaRequest(page: Int): Request = GET(baseUrl, headers)
+
+    override fun popularMangaSelector(): String = "div.last-updates > div.item"
+
+    override fun popularMangaNextPageSelector(): String? = null
+
+    override fun popularMangaFromElement(element: Element): SManga = SManga.create().apply {
+        with(element.selectFirst("div.panel-body div.mangaThumbnail")!!) {
+            val mangaUrl = selectFirst("a")!!.attr("href")
+                .removeSuffix("/")
+                .substringBeforeLast("/")
+                .replaceFirst("/reader/", "/plot/manga/")
+            setUrlWithoutDomain(mangaUrl)
+            thumbnail_url = select("img").imgAttr()
+            title = select("img").attr("title")
+        }
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request = throw UnsupportedOperationException()
+
+    override fun latestUpdatesSelector(): String = throw UnsupportedOperationException()
+
+    override fun latestUpdatesNextPageSelector(): String? = null
+
+    override fun latestUpdatesFromElement(element: Element): SManga = throw UnsupportedOperationException()
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/wp-admin/admin-ajax.php"
+
+        val ajaxHeaders = this.headersBuilder()
+            .add("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8")
```

Comment:
afaik okhttp will automatically set Content-Type header based on the request body